pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1772,13 +1772,12 @@ public void handleCommandStatus(String status, int updateCount, long insertOID)
           PSQLState.PROTOCOL_VIOLATION));
     }
 
-    public void handleCompletion() throws SQLException {
-      SQLWarning warning = getWarning();
-      if (warning != null) {
-        PgResultSet.this.addWarning(warning);
-      }
-      super.handleCompletion();
+    @Override
+    public void handleWarning(SQLWarning warning) {
+      super.handleWarning(warning);
+      PgResultSet.this.warnings = this.getWarning();
```

Comment:
@vlsi I decided to revert the change to `ResultSet` because I couldn't actually come up with test a scenario where it made sense.  
I'm not sure it is even possible for postgres to raise warnings while a result set is reading from a cursor.